File: complete/src/main/java/com/example/caching/CachingApplication.java
package com.example.caching;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.ComponentScan;

@SpringBootApplication
@ComponentScan({ "com.example.*" })
public class CachingApplication {

	public static void main(String[] args) {
		SpringApplication.run(CachingApplication.class, args);
	}

}

File: complete-kotlin/src/main/kotlin/com/example/caching/CachingApplication.kt
package com.example.caching

import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.cache.annotation.EnableCaching

@SpringBootApplication
@EnableCaching
class CachingApplication

fun main(args: Array<String>) {
    runApplication<CachingApplication>(*args)
}
